[SPARK-37513][SQL][DOC] date +/- interval with only day-time fields returns different data type between Spark3.2 and Spark3.1 (#34766)
apache/spark: docs/sql-migration-guide.md

  - In Spark 3.2, create/alter view will fail if the input query output columns contain auto-generated alias. This is necessary to make sure the query output column names are stable across different spark versions. To restore the behavior before Spark 3.2, set `spark.sql.legacy.allowAutoGeneratedAliasForView` to `true`.

>   - In Spark 3.2, additive expression of date and interval like `date '2011-11-11' + interval 12 hours` will return timestamp. In Spark 3.1 and earlier, the same expression will return date. To restore the behavior before Spark 3.2, you can use `cast` to convert timestamp as date.

Reviewer: ```suggestion
  - In Spark 3.2, date +/- interval with only day-time fields such as `date '2011-11-11' + interval 12 hours` returns timestamp. In Spark 3.1 and earlier, the same expression returns date. To restore the behavior before Spark 3.2, you can use `cast` to convert timestamp as date.
```
